src/cmd/manage_nsfs.js

```diff
@@ -361,7 +416,37 @@ async function bucket_management(action, user_input) {
     } else if (action === ACTIONS.STATUS) {
         response = await get_bucket_status(data);
     } else if (action === ACTIONS.UPDATE) {
-        response = await update_bucket(data);
+        const bucket_path = config_fs.get_bucket_path_by_name(user_input.name);
+        const bucket_lock_file = `${bucket_path}.lock`;
+        await native_fs_utils.lock_and_run(config_fs.fs_context, bucket_lock_file, async () => {
```

Comment:
can you move await native_fs_utils.lock_and_run() to be before line 409? 
we tried to keep this function minimal and simple and with these changes it adds an inline complexity 
also, adding lock_and_run before all functions (except LIST) will be better as I see it, WDYT?

Reply:
I don't think even other ops would require locking but yeah, won't do any harm either. I will add it and hopefully that brings down the visual complexity.

Reply:
so at least mention it in the issue, we should think about it and since it'll be backported we shouldn't do this change in this PR

Reply:
Sure. I will mention it in the issue created for the previous comment.